### ISSUE TITLE
   [Error]: ENVIRONMENT_FALLBACK | code: 'ENVIRONMENT_FALLBACK', |   originalMessage: undefined

### ISSUE THREAD

**Issue by ZehuaZhang** [amannn/next-intl#993]
### Description

```
Generating static pages (0/384)  [    ]u [Error]: ENVIRONMENT_FALLBACK

  code: 'ENVIRONMENT_FALLBACK',
  originalMessage: undefined
```

### Verifications

- [X] I've verified that the problem I'm experiencing isn't covered in [the docs](https://next-intl-docs.vercel.app/).
- [X] I've searched for similar, existing issues on GitHub and [Stack Overflow](https://stackoverflow.com/search?q=next-intl).
- [X] I've compared my app to [a working example](https://next-intl-docs.vercel.app/examples) to look for differences.

### Mandatory reproduction URL

hubnx.com

### Reproduction description

`next build`

### Expected behaviour

no error

**Comment by amannn** [comment]
Closing since no reproduction was provided and the problem is very likely in the app code.

**Comment by ZehuaZhang** [comment]
> Closing since no reproduction was provided and the problem is very likely in the app code.

Could you give me some suggestions? I followed your page router guide. Dev is running okay, but the production build gave me error for two days， so I have to temp change to react-intl to keep my production website running.

Do you have slack or email so we could discuss. I really like this easy to use library, unfortunately I cannot provide code here.

TLDR: 1. Followed your page router setup 2. Use getStaticProps in most pages 3. The rest pages already using getServerProps so I use getServerSideProps instead to get messages

I running into problems of two `u [Error]: ENVIRONMENT_FALLBACK` and `u [Error]: MISSING MESSAGES`, but `intl-ally` vscode extension checked all languages all translated, and running in local dev is okay.

Could you at least tell me what are those error telling me about?

**Comment by amannn** [comment]
Typically, `npm run build` should include which page has caused the error when being pre-rendered. I'd start there and navigate to the corresponding page in development to get more details about the error.

The errors indicate:
- `ENVIRONMENT_FALLBACK`: You're using either date/time formatting and no `timeZone` was configured on the provider, or you're using relative date/time formatting and no `now` value was configured.
- `MISSING MESSAGES`: No messages are available on the provider for the given page (see e.g. https://github.com/amannn/next-intl/issues/87#issuecomment-1243349403).

As mentioned above, you should get more details during development, so that should be your first step with debugging.

Hope this helps!

**Comment by ZehuaZhang** [comment]
@amannn Thank you, I will try those later. For page router, do you have an example to initialize it in a central place, e.g. `app.getInitialProps`, instead of `getStaticProps` and `getServerSideProps` on a per-page level?

I've looked at your page router examples, and didn't find one with initializing in one place for page router, so just wondering if you have one by chance.

**Comment by amannn** [comment]
You can init in `App`, yes, I don’t have an example for that currently though.

**Comment by aelabid** [comment]
I faced the same problem and adding `timeZone` fixed the build problem for me too
